sampletracksapp/src/main/java/com/example/sampletracksapp/ExperimentationDialogFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.example.sampletracksapp
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import com.automattic.android.experimentation.ExPlat
+import com.automattic.android.experimentation.Experiment
+import com.automattic.android.experimentation.ExperimentLogger
+import com.example.sampletracksapp.databinding.DialogExperimentationBinding
+import kotlinx.coroutines.GlobalScope
+import java.io.File
+import java.util.UUID
+
+class ExperimentationDialogFragment : DialogFragment() {
+
+    private var exPlat: ExPlat? = null
+
+    override fun onStart() {
+        super.onStart()
+        dialog?.let {
+            val width = ViewGroup.LayoutParams.MATCH_PARENT
+            val height = ViewGroup.LayoutParams.MATCH_PARENT
+            it.window?.setLayout(width, height)
+        }
+    }
+
+    override fun onCreateView(
```

Comment:
FYI: This is the second time I write this comment (😭) as I forgot to save it, then submitted the review without saving it, which caused me to lose it... 😞 

-----

Question (❓): While testing this I was only getting assignments with an empty list of `variations`. I remember you showcasing that to us the other day and that it works, it is just that I couldn't make it show assignments with `variations`. I used the following:

- Platform to query: `wpandroid`
- Experiment names: Copied the `name` from [ExPlat](https://experiments.a8c.com/experiments)
- Anon ID: Tried with and without that
- How: First clicked `SETUP` then `FETCH`

No sure if I am just unlucky, doing something wrong, or it not working as I expected... 🤷 

-----

`EXTRAS`:

1. Suggestion (💡): To show the output of the result of our actions in a `TextView` below the buttons, instead of jumping to `LogCat`, just like it is done on `FluxC`, and help testers test.
2. Suggestion (💡): To make `Platform to Query` mandatory before hitting `SETUP`, thus disabling this button at first. Same applies to `Experiment names`.
3. Suggestion (💡): To mark `Anon ID` optional so that testers know this is not required.
4. Suggestion (💡): To disable the `FETCH` button until `SETUP` is triggered.
5. Suggestion (💡): To move the `GET CACHE` and `CLEAR CACHE` button on a separate row, below the `FETCH` and `SETUP` button, them being a "somehow" secondary testing buttons.

PS: Please ignore any of those above, I just wanted to put those idea here for you, just in case you would have liked applying some. We can also do all that later-on.

Reply:
> Platform to query

You have to choose the correct platform - I see on ExPlat there's no experiments for `wpandroid` right now. It's a third column on the table :)

Reply:
Ah @wzieba , you're right, I was looking for this but within a specific experiment page (ie. 21918-explat-experiment) and couldn't find it... It was in front of my eyes and you also showed that to us! 😭

FYI: Btw, I just tested that again with a few such experiment names, using the correct platform name this time and I never got an `assignment` with a non-empty list of `variations`... 🤷

Reply:
> FYI: Btw, I just tested that again with a few such experiment names, using the correct platform name this time and I never got an assignment with a non-empty list of variations... 🤷

This can be the case of an experiment not working with `anonId`, but maybe with a bearer token identification. I've asked a clarifying question internally, p1725007206945009-slack-C7HH3V5AS

But for now, please check the currently running `calypso` experiment with higher number of participants - this one should return some variations (I just tested it)

Reply:
> EXTRAS:

Initially, I wasn't sure if it's worth investing time into this, but maybe you're right - better example app maybe can help with understanding the SDK in the future.

I've applied all the suggestions, take a look!

Reply:
👋 @wzieba and thanks for the reply!

> This can be the case of an experiment not working with anonId, but maybe with a bearer token identification. I've asked a clarifying question internally, p1725007206945009-slack-C7HH3V5AS
> 
> But for now, please check the currently running calypso experiment with higher number of participants - this one should return some variations (I just tested it)

Thanks for the message on Slack! 🙇 

Btw, I did try with and without the `anonId` yesterday, and did try this `calypso` experiment you suggested, did work, was only getting assignments without any variations... 🤷 

But, I just tried again, and although the first time, without `anonId` I wasn't getting any variations, when I tried a second time and SETUP the experiment with an `anonId` it start working, the first time I see an assignment with a variation: 🎉 

`variations={calypso_disable_site_assembler=Treatment(name=treatment_disable_onboarding)}`

So, all good I think, I was just unlucky (I guess) yesterday all day, not being able to get any variations... 😅 

-----

> Initially, I wasn't sure if it's worth investing time into this, but maybe you're right - better example app maybe can help with understanding the SDK in the future.
> 
> I've applied all the suggestions, take a look!

Yea, I only suggested those because I thought them to be quick wins, thanks for applying the suggestion, taking a look se speak! 🥇

Reply:
This awesome @wzieba , I just tested the improvements and that looks so much better! 🥇 

Few extra quick suggestions to take this to 100%:
1. Making disabling the `GENERATE` too when `SETUP` hasn't happened yet. This is because of the `exPlat?.clear()` which will not work when hitting generate, but maybe someone will expect that. PS: Same apply for `CLEAR CACHE` because maybe it will become misleading/frustrating to testers thinking this works. Only `GET CACHE` does not depend on `exPlat` being initialized.
2. Adding the result of `GET CACHE` to the output too. This way it will become evident whether `CLEAR CACHE` worked or not.
3. When `GET CACHE` doesn't exist, let's output that too, both on screen and in the `LogCat` logs.

Reply:
> Thanks, good suggestions 👍 I applied all the fixes.

This is awesome, I never had to open the `LogCat` during this test iteration!

> And I found an issue, that addressing might help in testing: generateAnonId wasn't providing the anonymous id to ExPlat 🫣 Fixed in https://github.com/Automattic/Automattic-Tracks-Android/commit/17fd0e05b2697048015f9a8915b9c7289fa51afe

True, when reviewing I was thinking that you expect testers to hit `SETUP` afterwards, but with this change this is not needed! 💯 

> This proves though, that anonId is not a requirement for some experiments.

I guess so! 😅 

-----

This too I think it is ready to merge! 🚀

Reply:
> True, when reviewing I was thinking that you expect testers to hit SETUP afterwards, but with this change this is not needed! 💯

I think this was initially my idea, but I think it was not very clear (it wasn't for me after 1 day 😄 )

Reply:
> (it wasn't for me after 1 day 😄 )

Then it is 100% the right call! 🤣 🤣 🤣